developer/src/kmc-ldml/src/compiler/compiler.ts

```diff
@@ -41,29 +41,74 @@ export const SECTION_COMPILERS = [
   TranCompiler,
 ];
 
+/**
+ * @public
+ * Internal in-memory build artifacts from a successful compilation
+ */
 export interface LdmlKeyboardCompilerArtifacts extends KeymanCompilerArtifacts {
+  /**
+   * Binary keyboard filedata and filename - installable into Keyman desktop
+   * projects
+   */
   kmx?: KeymanCompilerArtifactOptional;
+  /**
+   * Binary on screen keyboard filedata and filename - installable into Keyman
+   * desktop projects alongside .kmx
+   */
   kvk?: KeymanCompilerArtifactOptional;
+  /**
+   * Javascript keyboard filedata and filename - installable into KeymanWeb,
+   * Keyman mobile products
+   */
   js?: KeymanCompilerArtifactOptional;
 };
 
 export interface LdmlKeyboardCompilerResult extends KeymanCompilerResult {
+  /**
+   * Internal in-memory build artifacts from a successful compilation. Caller
+   * can write these to disk with {@link LdmlKeyboardCompiler.write}
+   */
   artifacts: LdmlKeyboardCompilerArtifacts;
 };
 
+/**
+ * @public
+ * Compiles a LDML keyboard .xml file to a .kmx (KMXPlus), .kvk, and/or .js. The
+ * compiler does not read or write from filesystem or network directly, but
+ * relies on callbacks for all external IO.
+ */
 export class LdmlKeyboardCompiler implements KeymanCompiler {
   private callbacks: CompilerCallbacks;
   private options: LdmlCompilerOptions;
 
   // uset parser
   private usetparser?: UnicodeSetParser = undefined;
 
+  /**
+   * Initialize the compiler, including loading the WASM host for uset parsing.
+   * Copies options.
+   * @param callbacks - Callbacks for external interfaces, including message
+   *                    reporting and file io
+   * @param options   - Compiler options
+   * @returns           false if initialization fails
+   */
   async init(callbacks: CompilerCallbacks, options: LdmlCompilerOptions): Promise<boolean> {
```

Comment:
are these auto-generated? Noticing it is inserthing a hyphen after the param names and before the description but no hyphen on the @returns. Wondering if it is an intential change.

Reply:
The hyphen is optional for jsdoc and required for tsdoc, it seems.

> The @param tag is followed by a parameter name, followed by a hyphen, followed by a description.

https://tsdoc.org/pages/tags/param/